go/worker/common/p2p/crypto.go

```diff
@@ -28,6 +28,10 @@ func (s *p2pSigner) Equals(other libp2pCrypto.Key) bool {
 }
 
 func (s *p2pSigner) Raw() ([]byte, error) {
+	if us, ok := s.signer.(signature.UnsafeSigner); ok {
```

Comment:
Instead of doing this, maybe generate and persist an appropriate amount of system entropy or something?